src/main/java/org/truffleruby/core/regexp/ClassicRegexp.java

```diff
@@ -942,13 +943,15 @@ public RopeBuilder toRopeBuilder() {
     }
 
     @TruffleBoundary
-    public void appendRegexpString19(RopeBuilder to, Rope str, int start, int len, Encoding resEnc) {
-        int p = start;
+    public void appendRegexpString19(RopeBuilder to, TStringWithEncoding str, int start, int len, Encoding resEnc) {
+        var byteArray = str.getInternalByteArray();
+        int p = start + byteArray.getOffset();
```

Comment:
`to.append(bytes, start, len);` below doesn't seem right together with this line.
It looks like appendRegexpString19() is already called with start=p from the caller, so probably it should `int p = start`.
I think what we should is pass the InternalByteArray explicitly to this method, so it's clearer it's using the same byte array and the offset is already included in the passed `start` argument.